images/access-setup/content/bin/setup_compute.sh

```diff
@@ -136,6 +138,38 @@ check_prerequisites() {
   fi
 }
 
+generate_shared_manifests(){
+  printf "[Shared manifests]\n"
+
+  printf "    - tekton-chains signing key: "
+  manifest="$manifests_dir/compute/tekton-chains/signing-secrets.yaml"
+  if [ ! -e "$manifest" ]; then
+    printf "\n"
+    manifests_tmp_dir="$(dirname "$manifest")/tmp"
+    mkdir -p "$manifests_tmp_dir"
+    cosign_passwd="$( head -c 12 /dev/urandom | base64 )"
+    echo -n "$cosign_passwd" > "$manifests_tmp_dir/cosign.password"
+    cosign_image="quay.io/redhat-appstudio/appstudio-utils:eb94f28fe2d7c182f15e659d0fdb66f87b0b3b6b"
+    podman run \
+      --rm \
+      --env COSIGN_PASSWORD="$cosign_passwd" \
+      --volume "$manifests_tmp_dir":/workspace:z \
+      --workdir /workspace \
+      --entrypoint /usr/bin/cosign \
+      "$cosign_image" generate-key-pair
+    {
+      echo "---"
+      kubectl create namespace tekton-chains --dry-run=client -o yaml
+      echo "---"
+      kubectl create secret generic -n tekton-chains signing-secrets --from-file="$manifests_tmp_dir" --dry-run=client -o yaml | \
+        yq '. += {"immutable" :true}' | \
+        yq "sort_keys(.)"
+    } > "$manifest"
```

Comment:
The results (positive or negative) of the previous --dry-run commands are stored in $manifest but doesn't look like we are checking anything after that.

Reply:
That file will be deployed by the `install_shared_manifests` function in `images/cluster-setup/bin/install.sh`.

Please close this conversation if that answers your question. If not please reformulate your question.

Reply:
I've added a safeguard during the creation of the manifest.